src/zjs_callbacks.c

```diff
@@ -30,33 +30,33 @@
 
 #define CALLBACK_TYPE_JS    0
 #define CALLBACK_TYPE_C     1
+#define JS_TYPE_SINGLE      0
+#define JS_TYPE_LIST        1
```

Comment:
You never seem to use this JS_TYPE_LIST, and you use the JS_TYPE_SINGLE once but don't explicitly set it anywhere, need to make more use of these I think to distinguish which value in the union you should be looking for.